extension/js/common/platform/store/acct-store.ts

```diff
@@ -125,19 +129,30 @@ export class AcctStore extends AbstractStore {
   };
 
   public static getScopes = async (acctEmail: string): Promise<Scopes> => {
-    const { google_token_scopes } = await AcctStore.get(acctEmail, ['google_token_scopes']);
+    const accessToken = await InMemoryStore.get(acctEmail, InMemoryStoreKeys.GOOGLE_TOKEN_ACCESS);
+    // const { google_token_scopes } = await AcctStore.get(acctEmail, ['google_token_scopes']);
     const result: { [key in GoogleAuthScopesNames]: boolean } = {
       email: false, openid: false, profile: false, compose: false,
       modify: false, readContacts: false, readOtherContacts: false, gmail: false
     };
-    if (google_token_scopes) {
-      for (const key of Object.keys({ ...GoogleAuth.OAUTH.scopes, ...GoogleAuth.OAUTH.legacy_scopes })) {
-        const scopeName = key as GoogleAuthScopesNames;
-        if (scopeName in GoogleAuth.OAUTH.scopes) {
-          result[scopeName] = google_token_scopes.includes(GoogleAuth.OAUTH.scopes[scopeName as keyof typeof GoogleAuth.OAUTH.scopes]);
-        } else if (scopeName in GoogleAuth.OAUTH.legacy_scopes) {
-          result[scopeName] = google_token_scopes.includes(GoogleAuth.OAUTH.legacy_scopes[scopeName as keyof typeof GoogleAuth.OAUTH.legacy_scopes]);
-        }
+    if (!accessToken) {
+      return result;
+    }
+    let allowedScopes: string[] = [];
+    try {
+      const { scope } = await Api.ajax({ url: `https://www.googleapis.com/oauth2/v1/tokeninfo?access_token=${accessToken}` }, Catch.stackTrace()) as { scope: string };
```

Comment:
Should use `OAUTH_GOOGLE_API_HOST` (preferred)
or `GMAIL_GOOGLE_API_HOST` so this doesn't make its way to mock builds